server/models/Pet.js

```diff
@@ -0,0 +1,47 @@
+import pg from "pg"
+
+const pool = new pg.Pool({
+  connectionString: "postgres://postgres:password@localhost:5432/pet_adoption_agency"
+})
+
+class Pet {
+  constructor({
+    id,
+    name,
+    img_url,
+    imgUrl,
+    age,
+    vaccination_status,
+    vaccinationStatus,
+    adoption_story,
+    adoptionStory,
+    available_for_adoption,
+    availableForAdoption,
+    pet_type_id,
+    petTypeId
+  }) {
+    this.id = id
+    this.name = name
+    this.imgUrl = img_url || imgUrl
+    this.age = age
+    this.vaccinationStatus = vaccination_status || vaccinationStatus
+    this.adoptionStory = adoption_story || adoptionStory
+    this.availableForAdoption = available_for_adoption || availableForAdoption
+    this.petTypeId = pet_type_id || petTypeId
+  }
+
+  static async findById(id) {
+    try {
+      const result = await pool.query("SELECT * FROM pets WHERE pet_type_id= $1", [id])
+      const petByTypeData = result.rows
+      const petsByType = petByTypeData.map(petByType => new Pet(petByType))
+
+      return petsByType
+    } catch (err) {
+      console.error(err)
+      throw err
+    }
+  }
```

Comment:
This method should not be in your `Pet` model! `Pet.findById` should be finding a _pet_ from the _pets_ table with a given id -- not from the `pet_types` table.

Move this method over to your `PetType` model. It will look pretty much exactly the same, you'll just need to import the Pet class inside that model!